src/server/classroom/message/message.controller.js

```diff
@@ -0,0 +1,97 @@
+const httpStatus = require('http-status');
+const cachegoose = require('cachegoose');
+const Classroom = require('../../models/classroom.model');
+const Messages = require('../../models/message.model');
+const APIError = require('../../helpers/APIError');
+
+/**
+ *
+ * @param {[*, *]} packet
+ * @param {import('socket.io').Socket} socket
+ * @param {import('socket.io').Server} io
+ */
+async function sendMessage(packet, socket, io) {
+  const [data, callback, meta] = packet;
+
+  if (!meta.invokerClassroom) return callback({ error: 'You are not in a classroom' });
+  let classroom = meta.invokerClassroom;
+  const messageType = data.information.type;
+  let message;
+
+  switch (messageType) {
+    case 'text':
+      message = await Messages.TextMessage.create({
+        sender: meta.invoker.id,
+        type: messageType,
+        content: data.message,
+        classroom: classroom.id
+      }); break;
+    case 'question':
+      message = await Messages.QuestionMessage.create({
+        sender: meta.invoker.id,
+        type: messageType,
+        content: {
+          isResolved: false,
+          content: data.message
+        },
+        classroom: classroom.id
+      }); break;
+    case 'reply': {
+      classroom = await classroom.populate('messages').execPopulate();
```

Comment:
This is what's missing. The `messages` array in a classroom database object is only an array of message IDs, not message database objects. Using `populate` here tells mongoose to convert them all to message database objects instead, so you can access the message details